setup.py

```diff
@@ -92,92 +103,82 @@ def run(self):
+    # Significant effort was put in to trying to get these to compile
+    # within this build process but there was no obvious way of finding
+    # a workaround to the issue presented in
+    # https://github.com/SciTools/iris/issues/2481.
 
-    """
-    description = "generate CF standard name module"
-    user_options = []
+    shelled_code = textwrap.dedent("""\
 
-    def initialize_options(self):
-        pass
+    import os
+    from pyke import knowledge_engine
 
-    def finalize_options(self):
-        pass
+    # Monkey patch the load method to avoid "ModuleNotFoundError: No module
+    # named 'iris.fileformats._pyke_rules.compiled_krb'". In this instance
+    # we simply don't want the knowledge engine, so we turn the load method
+    # into a no-op.
+    from pyke.target_pkg import target_pkg
+    target_pkg.load = lambda *args, **kwargs: None
 
-    def run(self):
-        cmd = std_name_cmd('lib')
-        self.spawn(cmd)
+    # Compile the rules by hand, without importing iris. That way we can
+    # avoid the need for all of iris' dependencies being installed.
+    os.chdir(os.path.join('{bld_dir}', 'iris', 'fileformats', '_pyke_rules'))
 
+    knowledge_engine.engine('')
 
-class MakePykeRules(Command):
-    """
-    Compile the PyKE CF-NetCDF loader rule base.
+    """.format(bld_dir=directory)).split('\n')
+    shelled_code = '; '.join(
+        [line for line in shelled_code
+         if not line.strip().startswith('#') and line.strip()])
+    args = [sys.executable, '-c', shelled_code]
+    cmd.spawn(args)
 
-    """
-    description = "compile CF-NetCDF loader rule base"
-    user_options = []
 
-    def initialize_options(self):
-        pass
+def copy_copyright(cmd, directory):
+    # Copy the COPYRIGHT information into the package root
+    iris_build_dir = os.path.join(directory, 'iris')
+    for fname in ['COPYING', 'COPYING.LESSER']:
+        copyfile(fname, os.path.join(iris_build_dir, fname))
 
-    def finalize_options(self):
-        pass
 
-    @staticmethod
-    def _pyke_rule_compile():
-        """Compile the PyKE rule base."""
-        from pyke import knowledge_engine
-        import iris.fileformats._pyke_rules
-        knowledge_engine.engine(iris.fileformats._pyke_rules)
+def build_std_names(cmd, directory):
+    # Call out to tools/generate_std_names.py to build std_names module.
 
-    def run(self):
-        # Compile the PyKE rules.
-        MakePykeRules._pyke_rule_compile()
+    script_path = os.path.join('tools', 'generate_std_names.py')
+    xml_path = os.path.join('etc', 'cf-standard-name-table.xml')
+    module_path = os.path.join(directory, 'iris', 'std_names.py')
+    args = (sys.executable, script_path, xml_path, module_path)
 
+    cmd.spawn(args)
 
-class MissingHeaderError(Exception):
-    """
-    Raised when one or more files do not have the required copyright
-    and licence header.
 
+def custom_cmd(command_to_override, functions, help_doc=""):
```

Comment:
`command_to_override` -> `cls` ?

Reply:
Compromise? ```command_cls_to_override```?

Reply:
@pelson This change got lost in the noise ... although, I'm not going to lose any sleep if you don't do it 😜